src/agentscope/server/launcher.py

```diff
@@ -0,0 +1,464 @@
+# -*- coding: utf-8 -*-
+""" Server of distributed agent"""
+import os
+from multiprocessing import Process, Event, Pipe
+from multiprocessing.synchronize import Event as EventClass
+import socket
+import asyncio
+import signal
+import argparse
+from typing import Type, Optional
+from concurrent import futures
+from loguru import logger
+
+try:
+    import grpc
+    from agentscope.rpc.rpc_agent_pb2_grpc import (
+        add_RpcAgentServicer_to_server,
+    )
+except ImportError as import_error:
+    from agentscope.utils.tools import ImportErrorReporter
+
+    grpc = ImportErrorReporter(import_error, "distribute")
+    add_RpcAgentServicer_to_server = ImportErrorReporter(
+        import_error,
+        "distribute",
+    )
+
+import agentscope
+from agentscope.server.servicer import AgentServerServicer
+from agentscope.agents.agent import AgentBase
+from agentscope.utils.tools import _get_timestamp
+
+
+def setup_agent_server(
+    host: str,
+    port: int,
+    server_id: str,
+    init_settings: dict = None,
+    start_event: EventClass = None,
+    stop_event: EventClass = None,
+    pipe: int = None,
+    local_mode: bool = True,
+    max_pool_size: int = 8192,
+    max_timeout_seconds: int = 1800,
+    custom_agents: list = None,
+) -> None:
+    """Setup agent server.
+
+    Args:
+        host (`str`, defaults to `"localhost"`):
+            Hostname of the agent server.
+        port (`int`):
+            The socket port monitored by the agent server.
+        server_id (`str`):
+            The id of the server.
+        init_settings (`dict`, defaults to `None`):
+            Init settings for agentscope.init.
+        start_event (`EventClass`, defaults to `None`):
+            An Event instance used to determine whether the child process
+            has been started.
+        stop_event (`EventClass`, defaults to `None`):
+            The stop Event instance used to determine whether the child
+            process has been stopped.
+        pipe (`int`, defaults to `None`):
+            A pipe instance used to pass the actual port of the server.
+        local_mode (`bool`, defaults to `None`):
+            Only listen to local requests.
+        max_pool_size (`int`, defaults to `8192`):
+            Max number of task results that the server can accommodate.
```

Comment:
What means "task" here?

Reply:
![image](https://github.com/modelscope/agentscope/assets/102287034/771a27ee-4e1e-4df0-a0c9-3062ffeb51de)

The key concepts of actor model in wikipedia